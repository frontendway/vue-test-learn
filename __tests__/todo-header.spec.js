import { shallowMount } from '@vue/test-utils'
import TodoHeader from '../src/views/todo-list/components/todo-header.vue'

it('header 包含 input', () => {
  const wrapper = shallowMount(TodoHeader)
  const input = wrapper.find('input[type="text"]')
  expect(input.exists()).toBeTruthy()
})

it('header 中 input 初始值为空', () => {
  const wrapper = shallowMount(TodoHeader)
  const inputValue = wrapper.vm.value
  expect(inputValue).toBe('')
})

it('header 中 input 发生变化数据也发生变化', () => {
  const wrapper = shallowMount(TodoHeader)
  const input = wrapper.find('input[type="text"]')
  input.setValue('new text')
  expect(wrapper.vm.value).toBe('new text')
})

it('header 中 input 回车，无内容时无反应', () => {
  const wrapper = shallowMount(TodoHeader)
  const input = wrapper.find('input[type="text"]')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})

it('header 中 input 回车，有内容则向外派发 add 事件，且清空 input', () => {
  const wrapper = shallowMount(TodoHeader)
  const input = wrapper.find('input[type="text"]')
  input.setValue('text')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  expect(wrapper.vm.value).toBe('')
})

it('header 样式改变做提示', () => {
  const wrapper = shallowMount(TodoHeader)
  expect(wrapper).toMatchSnapshot()
})
