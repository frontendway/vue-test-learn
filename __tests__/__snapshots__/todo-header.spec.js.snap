// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header 样式改变做提示 1`] = `<header class="todo-header"><input type="text" data-test="input"></header>`;
